### PR TITLE
Move holdReady to deprecated

### PR DESCRIPTION
**Pull request by kumarmj**
fixes issue #3288

### CODE REVIEW

**Comment by timmywil** [comment]
LGTM

**Comment by mgol** [comment]
LGTM

File: src/core/ready.js [hunk 1]
```diff
@@ -34,15 +34,6 @@ jQuery.extend( {
 	// the ready event fires. See #6781
```
Comment (dmethvin) [inline src/core/ready.js]: When removing this we may want to note in a comment that Migrate needs `readyWait` to shim `.holdReady()`. If/when .holdReady()` is removed in v4 we might be tempted to remove or simplify this logic, or make `readyWait` private. A similar thing happened in 3.0 when we removed `jQuery.sub()` and simplified `jQuery.fn.init()` so I'm just being paranoid.

**Comment by dmethvin** [comment]
We have a `.holdReady()` test but it doesn't look like it runs as part of the unit tests? I was going to ask whether anything needed to be moved to `test/deprecated.js` but it doesn't look like it does.

**Comment by dmethvin** [comment]
Oh! See gh-3331.

**Pull request closed without merging**